tests/unit/FileHelperTest.php

```diff
@@ -0,0 +1,325 @@
+        $this->assertEquals( $expected, $actual );
+    }
+
+    /**
+     * Test filePathLooksCrawlable method's wp2static_filenames_to_ignore
+     * filter.
+     *
+     * @return void
+     */
+    public function testFilePathLooksCrawlableFilenameFilter() {
+        // thumbs.db filenames are currently disallowed
+        $expected = false;
+        $actual = FilesHelper::filePathLooksCrawlable( "/path/to/thumbs.db" );
+        $this->assertEquals( $expected, $actual );
+
+        // Here we're changing which fil eextensions are no longer allowed.
+        \WP_Mock::onFilter( 'wp2static_filenames_to_ignore' )
+            ->with([
+                '__MACOSX',
+                '.babelrc',
+                '.gitignore',
+                '.gitkeep',
+                '.htaccess',
+                '.php',
+                '.travis.yml',
+                'backwpup',
+                'bower_components',
+                'bower.json',
+                'composer.json',
+                'composer.lock',
+                'config.rb',
+                'current-export',
+                'Dockerfile',
+                'gulpfile.js',
+                'latest-export',
+                'LICENSE',
+                'Makefile',
+                'node_modules',
+                'package.json',
+                'pb_backupbuddy',
+                'plugins/wp2static',
+                'previous-export',
+                'README',
+                'static-html-output-plugin',
+                'thumbs.db',
+                'tinymce',
+                'wc-logs',
+                'wpallexport',
+                'wpallimport',
+                'wp-static-html-output', // exclude earlier version exports
+                'wp2static-addon',
+                'wp2static-crawled-site',
+                'wp2static-processed-site',
+                'wp2static-working-files',
+                'yarn-error.log',
+                'yarn.lock',
+            ])
+            ->reply(['yarn.lock']);
+        // We've disallowed yarn.lock - test it
+        $expected = false;
+        $actual = FilesHelper::filePathLooksCrawlable( "/path/to/yarn.lock" );
+        $this->assertEquals( $expected, $actual );
+
+        // thumbs.db filenames should now be allowed - test it
+        $expected = true;
+        $actual = FilesHelper::filePathLooksCrawlable( "/path/to/thumbs.db" );
+        $this->assertEquals( $expected, $actual );
+    }
+
+    public function testCleanDetectedURLs() {
+        // Mock the WP functions used by FilesHelper::cleanDetectedURLs()
+        $mock = \Mockery::mock('alias:WP2Static\SiteInfo')
+            ->shouldReceive('getUrl')
+            ->andReturn('https://foo.com/');
+
+        // No trailing slash
+        $expected = ['/foo'];
+        $actual = FilesHelper::cleanDetectedURLs(["https://foo.com/foo"]);
+        $this->assertEquals( $expected, $actual );
+
+        // Trailing slash
+        $expected = ['/foo/'];
```

Comment:
Nice, thanks for including trailing/untrailing - I was unsure how the difference is affecting various areas of code at present!